src/services/AsyncValidationService.ts

```diff
@@ -0,0 +1,41 @@
+import express from 'express';
+import NodegenRequest from '@/http/interfaces/NodegenRequest';
+
+class AsyncValidationService {
+  /**
+   * Entry function for the src/http/nodegen/middleware/asyncValidationMiddleware.ts
+   */
+  middleware (req: NodegenRequest, res: express.Response, next: express.NextFunction, asyncValidators: string[]) {
+    this.parseValidators(req, asyncValidators).then(() => {
+      next();
+    }).catch((e) => {
+      throw e;
```

Comment:
I think this should be `next(e)`, otherwise you just get an unhandled rejection error

Reply:
ah really.. thought it bubbled back up to the promise router error handler.. will check it out and update